hub-cloud-upload/hub-cloud-upload.yaml

```diff
@@ -0,0 +1,51 @@
+name: Upload hub data to cloud storage
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  # Hubverse AWS account number
+  AWS_ACCOUNT: 767397675902
+
+permissions:
+  contents: read
+  # id-token write required for AWS auth
+  id-token: write
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Get hub cloud config
+      # save cloud-related fields from admin config as environment variables
+      # (jq json parser is installed on Github-hosted runners)
+      run: |
+        cloud_enabled=$(cat ./hub-config/admin.json | jq -r '.cloud.enabled') && echo "CLOUD_ENABLED=$cloud_enabled"
+        cloud_host=$(cat ./hub-config/admin.json | jq -r '.cloud.host.name') && echo "CLOUD_HOST=$cloud_host"
+        cloud_storage=$(cat ./hub-config/admin.json | jq -r '.cloud.host.storage') && echo "CLOUD_STORAGE=$cloud_storage"
+        echo "CLOUD_ENABLED=$cloud_enabled" >> $GITHUB_ENV
+        echo "CLOUD_HOST=$cloud_host" >> $GITHUB_ENV
+        echo "CLOUD_STORAGE=$cloud_storage" >> $GITHUB_ENV
+
+    - name: Configure AWS credentials
+      # request credentials to assume the hub's AWS role via OpenID Connect
+      if: env.CLOUD_ENABLED == 'true'
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT }}:role/${{ env.CLOUD_STORAGE }}
+        aws-region: us-east-1
+
+    - name: Sync files to cloud storage
+      if: env.CLOUD_ENABLED == 'true'
+      run: |
+        aws s3 sync ./hub-config "s3://$CLOUD_STORAGE/hub-config" --delete
```

Comment:
@nickreich @elray1 Should we include the optional hub directories on the list of folders to sync, or let hub admins update the GitHub action if they want to sync them? 

* `model-abstracts`
* `target-data`
* `auxiliary-data`

It would be possible to construct some sync commands that will only run if an optional folder exists, but it's unclear if we want that as a default behavior.

Reply:
That's a good question, 
I wonder if it make sense to have them syncs if they exist, but with the possibility to choose not to have them sync if we don't want to. 
For the `model-abstracts`, it might be interesting to have the possibility to sync them to the S3, but I think should be optional. For the US SMH hub, we display them on the website, so the information is already easily accessible but read-only version, so some persons might be interesting to easily access them.

Reply:
Thanks for the info @elray1 and @LucieContamin. Is this a correct summary?

1. for `target-data` and `auxiliary-data` folders: if the folder exists, sync it (the sync command will be a no-op if the folder is empty)
2. for `model-abstracts`: even if the folder exists, the sync should be optional

Before we introduce another config option or control variable to handle `model-abstracts`, I'm curious about the motivation for making the sync optional (I'm assuming this data isn't too large/costly, based on the [definition here](https://hubdocs.readthedocs.io/en/latest/user-guide/model-abstracts.html)).

Happy to do the optional `model-abstracts` logic if it's critical, just wanted to make sure we weren't making the decision based on perceived AWS cost, since introducing special handling by directory incurs its own set of maintainability costs.

Reply:
It's correct. I am debating if we also need to allow the `auxiliary-data` to be "optionally" synced. For example, for the SMH, some might be quite voluminous and use only for a specific round, so I don't know if it make sense to keep it sync when it's not used and/or if the size of the folder might be an issue in some cases. 

For the model-abstracts, it's because I was not sure if it's ever going to be accessed via the S3 bucket as they are available on the website.

Reply:
Is there more to making something optional than saying "you can delete this folder from a specification in the workflow definition file"?  If not, maybe we just add them all in here and document how users can update it (and/or maybe later on, add args to hubCI functionality to set these things)?

Reply:
@elray1 not really, unless we wanted to get fancy and make some kind of "directories to sync" option in `admin.json`

Which doesn't seem to fit the "do not change `admin.json` once a hub is launched" directive. So agree, let's keep it simple.

What do you and @LucieContamin think of this update that lists each folder to sync and has note to delete whatever you don't want on S3?

https://github.com/Infectious-Disease-Modeling-Hubs/hubverse-actions/pull/11/commits/cd9f7ad620074c8ba69e198475852e7d364451ec

Reply:
It looks good, thank you! I agree to keep it simple

Reply:
looks good to me too